crates/types/src/entities/relayer/transaction.rs

```diff
@@ -0,0 +1,123 @@
+//! Relayed (forced) transaction entity types
+
+use crate::{
+    blockchain::primitives::DaBlockHeight,
+    fuel_crypto,
+    fuel_types::Bytes32,
+};
+
+/// Transaction sent from the DA layer to fuel by the relayer
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum RelayedTransaction {
+    /// V1 version of the relayed transaction
+    V1(RelayedTransactionV1),
+}
+
+impl Default for RelayedTransaction {
+    fn default() -> Self {
+        Self::V1(Default::default())
+    }
+}
+
+/// The V1 version of the relayed transaction
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Default, Debug, Clone, PartialEq, Eq)]
+pub struct RelayedTransactionV1 {
+    /// The max gas that this transaction can consume
+    pub max_gas: u64,
+    /// The serialized transaction transmitted from the bridge
+    pub serialized_transaction: Vec<u8>,
+    /// The block height from the parent da layer that originated this transaction
+    pub da_height: DaBlockHeight,
+}
+
+/// The hash of a relayed transaction
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(
+    Debug,
+    Clone,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    derive_more::Display,
+    derive_more::From,
+    derive_more::Into,
+)]
+pub struct RelayedTransactionId(Bytes32);
+
+impl RelayedTransaction {
+    /// Get the DA height that originated this transaction from L1
+    pub fn da_height(&self) -> DaBlockHeight {
+        match self {
+            RelayedTransaction::V1(transaction) => transaction.da_height,
+        }
+    }
+
+    /// Set the da height
+    pub fn set_da_height(&mut self, height: DaBlockHeight) {
```

Comment:
For this setter, and all other setters in this interface, we should mark these with `#[cfg(test)]` or `#[cfg(any(test, feature = "test-helpers"))]` since they should only be used in test.

Reply:
done